src/components/QuickGrid/GroupByToolbar.tsx

```diff
@@ -0,0 +1,153 @@
+import * as React from 'react';
+import { GridColumn, SortDirection } from './QuickGrid.Props';
+import { Icon } from '../Icon/Icon';
+import * as _ from 'lodash';
+import { DropTarget } from 'react-dnd';
+import * as classNames from 'classnames';
+import { HeaderColumn, IHeaderColumnProps } from './HeaderColumn';
+import './QuickGrid.scss';
+
+export interface IGroupByProps {
+    groupBy: Array<string>;
+    columns: Array<GridColumn>;
+    onGroupByRemoved: (groupName: string) => void;
+    onGroupByChanged: (newGroupBy: Array<string>) => void;
+    sortColumn?: string;
+    sortDirection?: SortDirection;
+    onSort: (sortBy: string, sortDirection: SortDirection) => void;
+
+    // Drag&Drop props
+    canDrop?: any;
+    isOver?: any;
+    connectDropTarget?: any;
+}
+export interface IGroupByState {
+    groupBy: Array<string>;
+}
+
+class GroupByToolbarInner extends React.PureComponent<IGroupByProps, IGroupByState> {
+    constructor(props: IGroupByProps) {
+        super(props);
+        this.state = {
+            groupBy: props.groupBy
+        };
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (!_.isEqual(nextProps.groupBy, this.props.groupBy)) {
```

Comment:
this performs deep comparison? although it should not be problem in this specific case, i think we are much safer with shallow compare in react life cycle events